collectives/api.py

```diff
@@ -126,8 +146,21 @@ def users():
         value = request.args.get(f"filters[{i}][value]")
         field = request.args.get(f"filters[{i}][field]")
 
-        query_filter = getattr(User, field).ilike(f"%{value}%")
-        query = query.filter(query_filter)
+        if field == "roles":
+            # if field is roles,
+            for role_filter in value.split("-"):
+                role_value = None if role_filter[1:] == "none" else role_filter[1:]
+                if role_filter[0] == "r":
+                    print(id, flush=True)
+                    query_filter = User.roles.any(
+                        Role.role_id == RoleIds.get(role_value)
+                    )
+                if role_filter[0] == "t":
+                    query_filter = User.roles.any(Role.activity_id == role_value)
```

Comment:
Ca ne correspond pas exactement au filtre demandé: par exemple, si on demande un "Responsable d'activité ski de rando" ca va retourner tous les utilisateurs qui sont responsables d'une activité et on un role pour le ski de rendo (par exemple quelqu'un qui est RA Trail et initiateur Ski de rando, mais pas RA ski de rando)

Reply:
Je ne reproduit pas ce comportement. Tu peux réessayer de ton coté?

Reply:
Les filtres sont bien en et, mais la requete construite est du type:
(a un role "Responsable d'activité") ET (a un role pour l'activité "Ski de rando")
mais les deux conditions ne s'appliquent pas forcément au meme role si l'utilisateur en a plusieurs.
La requête correcte serait:
(a un role "Reponsable d'activité" pour l'activité "Ski de rando"), i.e. le `and` doit être à l'intérieur du `any`, pas entre deux `any`

Reply:
Ok, c'est bon, je vois le problème! Bien vu ;). Je vais corriger

Reply:
@gdaviet j'ai fait les changements. Est-ce que tu peux valider mes modifications stp?

Reply:
Oui c'est tout bon!